---
title: "Figure 7 A and S10 Inr variants and paused Pol II half-lives"
author: "Wanqing Shao(was@stowers.org) "
output:
  pdf_document:
    toc: yes
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancypagestyle{plain}{\pagestyle{fancy}}
---

# Description

To test if initiator (Inr) sequence variants can contribute to paused Pol II stability, we separated Inr G and non-G variants in our core promoter combination analysis. 

# Enviroment setup

``` {r setup, message=FALSE, error=FALSE}
library(GenomicRanges, warn.conflicts=F)
library(magrittr)
library(Rmisc)
library(magrittr)
library(Biostrings)
library(BSgenome.Dmelanogaster.UCSC.dm3)

setwd("/data/analysis_code")
options(knitr.figure_dir = 
        "Figure7A_S10_inr_variants_and_paused_polii_halflives"
        )

source("shared_code/knitr_common.r")
source("shared_code/ggplot_common.r")
source("shared_code/granges_common.r")
source("shared_code/metapeak_common.r")
source("shared_code/sample_common.r")
```

# Analysis

## Checking core promoter elements

```{r}
half_life_df <- get(load("rdata/half_life_df.RData"))
tss <- get(load("rdata/dme_mrna_unique_tss.RData"))

find_motif <- function(motif_name, fb_t_id,  mismatch=0) {
    
    motif_info <- subset(promoter_table, name == motif_name)
    motif <- DNAString(motif_info$motif)
    up_dis <- motif_info$window_start
    down_dis <- motif_info$window_end
    
    gene_tss <- tss[tss$fb_t_id %in% fb_t_id]
    
    if(up_dis >= 0 & down_dis >=0){
      tss_r <- resize(gene_tss, down_dis, "start") %>%
               resize(., down_dis - up_dis, "end")
    }
    if(up_dis < 0 & down_dis >=0){
      tss_r <- resize(gene_tss, down_dis, "start") %>%
               resize(., abs(up_dis)+down_dis, "end")
    }
    if(up_dis < 0 & down_dis <0){
      tss_r <- resize(gene_tss, abs(up_dis), "end") %>%
               resize(., abs(up_dis)-abs(down_dis), "start")
    }
    
    promoter_seq <- getSeq(Dmelanogaster, tss_r)
    names(promoter_seq) <- tss_r$fb_t_id
    
    count_df <- vcountPattern(motif, promoter_seq, fixed = FALSE, 
                              min.mismatch = 0, max.mismatch = mismatch) %>%
                data.frame(fb_t_id = fb_t_id, count =.)
    
    count_df$count <- ifelse(count_df$count >0, T, F)
    colnames(count_df)[2] <- motif_name
    count_df
}

promoter_table <- read.table("promoter_elements.txt", header=T)
motifs <- c("TATA", "DPE", "MTE", "PB", "Inr")
half_life_tss <- tss[tss$fb_t_id %in% half_life_df$fb_t_id]

motif_list <- lapply(as.character(motifs), function(x){
  motif <- find_motif(motif_name=x, half_life_tss$fb_t_id, mismatch = 1)
  motif
  })

motif_df <- reshape::merge_recurse(motif_list)
all_info_df <- merge(half_life_df, motif_df)
all_info_df$half_life <- ifelse(all_info_df$half_life >=0 & all_info_df$half_life<= 60, 
                                all_info_df$half_life, 60)

new_info_df <- 
  with(all_info_df, data.frame(fb_t_id = fb_t_id, gene = gene,
                               half_life = half_life, TATA = TATA,
                               Inr = Inr, pausing_elements = DPE | MTE | PB))
```

### Mutually exclusive model

We put genes into the following group:

-TATA only 
-TATA + Inr_non_G
-TATA + Inr_G
-Pausing only
-Pausing + Inr_non_G
-Pausing + Inr_G
-TATA + pasuing
-TATA + pausing + Inr_non_G
-TATA + pausing + Inr_G
-Inr_non_G only
-Inr_G only

```{r inr_differences_boxplot_mutually_exclusive, fig.height=6, fig.width=8}

inr_G_region_seq <- resize(half_life_tss, 2, "start") %>%
  resize(., 1, "end") %>%
  getSeq(Dmelanogaster, .) %>%
  as.character()

inr_info_df <- data.frame(fb_t_id = half_life_tss$fb_t_id,
                          G_at_2 = inr_G_region_seq == "G") 

new_info_df <- merge(new_info_df, inr_info_df)
new_info_df$Inr_G <- new_info_df$Inr &  new_info_df$G_at_2
new_info_df$Inr_non_G <- new_info_df$Inr &  !new_info_df$G_at_2

tata <- subset(new_info_df, TATA & !(Inr | pausing_elements)) %>%
  data.frame(type = "TATA")
tata_inr_non_g <- subset(new_info_df, TATA & Inr_non_G  & ! pausing_elements)%>%
  data.frame(type = "TATA Inr non G")
tata_inr_g <- subset(new_info_df, TATA & Inr_G  & ! pausing_elements)%>%
  data.frame(type = "TATA Inr G")

tata_pausing <- subset(new_info_df, TATA & pausing_elements & !Inr)%>%
  data.frame(type = "TATA pausing")
tata_pausing_inr_non_g <- subset(new_info_df, TATA & pausing_elements & Inr_non_G) %>%
  data.frame(type = "TATA pausing Inr non G")
tata_pausing_inr_g <- subset(new_info_df, TATA & pausing_elements & Inr_G)%>%
  data.frame(type = "TATA pausing Inr G")

pausing <- subset(new_info_df,  pausing_elements & !(Inr|TATA))%>%
  data.frame(type = "pausing")
pausing_inr_non_g <- subset(new_info_df,  pausing_elements & Inr_non_G & !TATA)%>%
  data.frame(type = "pausing Inr non G")
pausing_inr_g <- subset(new_info_df,  pausing_elements & Inr_G & !TATA)%>%
  data.frame(type = "pausing Inr G")

inr_non_g <- subset(new_info_df,  Inr_non_G & !(pausing_elements | TATA)) %>%
  data.frame(type = "Inr non G")
inr_g <- subset(new_info_df,  Inr_G & !(pausing_elements | TATA)) %>%
  data.frame(type = "Inr G")

all_combined <- rbind(tata,pausing, inr_non_g, inr_g, tata_pausing, tata_inr_non_g, 
                      pausing_inr_non_g, tata_pausing_inr_non_g, 
                      tata_inr_g, pausing_inr_g, tata_pausing_inr_g)

half_life_boxplot <- function(combined_df, title){
  count_info <- table(combined_df$type) %>% as.data.frame()
  ggplot(combined_df, aes(x = type, y = half_life)) + 
  geom_boxplot() +
  geom_text(data = count_info, aes(x = count_info$Var1, label = Freq, y = 65), 
               position = position_dodge(width = .75), 
               show.legend = FALSE )+
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  ylab("Paused Pol II half-life (min)") +
  xlab("") +
  ggtitle(title)
}


half_life_boxplot(all_combined, 
        "Inr variants and paused Pol II half-life \n mutually exclusive")

pairwise_type <- unique(all_combined$type) %>% 
  as.character(.) %>%
  combn(., 2, simplify = F)

calc_pval <- function(type1, type2, df){
  type1_df <- subset(df, type == type1)
  type2_df <- subset(df, type == type2)
  pval = wilcox.test(type1_df$half_life, type2_df$half_life)$ p.value
  data.frame(compare = paste(type1, "vs.", type2),
             pval = pval)
}
pval_df <- lapply(pairwise_type, function(x)calc_pval(x[1], x[2], all_combined)) %>%
  do.call(rbind, .)
pander(pval_df)

write.csv(pval_df, file = "Inr_variant_halflife_wilcox_test_mutually_exclusive.csv")
```

```{r inr_differences_heatmap_mutually_exclusive, fig.height=6, fig.width=8}
info_list <- list(TATA_TATA = tata,
                  Inr.non.G_Inr.non.G = inr_non_g,
                  Inr.G_Inr.G = inr_g,
                  Pausing_Pausing = pausing,
                  TATA_Inr.non.G = tata_inr_non_g,
                  TATA_Inr.G = tata_inr_g,
                  Pausing_Inr.non.G = pausing_inr_non_g,
                  Pausing_Inr.G = pausing_inr_g,
                  TATA_Pausing = tata_pausing)

generate_pairwise_df <- function(info_list){
  pairwise_df <- lapply(names(info_list), function(x){
  info_df <- data.frame(motif1 = gsub("_.*", "", x),
                        motif2 = gsub(".*_", "", x),
                        half_life = median(info_list[[x]]$half_life),
                        count = nrow(info_list[[x]]))
  info_df
  }) %>% do.call(rbind, .)
  pairwise_df_flip <- pairwise_df
  pairwise_df_flip$motif1 <- pairwise_df$motif2
  pairwise_df_flip$motif2 <- pairwise_df$motif1

  pairwise_df <- rbind(pairwise_df,pairwise_df_flip ) %>% unique()
  pairwise_df
}


promoter_number_heatmap <- function(pairwise_df){
  ggplot(pairwise_df, aes(x = motif1, y = motif2, fill = count)) + 
  scale_fill_gradient(low="#ffe1e8", high= "#a0457e") + 
  geom_tile() + xlab("") + ylab("") +
  geom_text(aes(label=count), color="black", size=8) +
  scale_x_discrete(position = "top") + 
  scale_y_discrete(position = "right") + 
  ggtitle("Number of promoters") +
  theme(panel.border = element_rect( colour = "black", fill = NA, size=1))
}

half_life_heatmap <- function(pairwise_df){
  ggplot(pairwise_df, aes(x = motif1, y = motif2, fill = half_life)) + 
  scale_fill_gradient(low="moccasin", high= "firebrick3") + 
  geom_tile() + xlab("") + ylab("") +
  geom_text(aes(label=round(half_life, digits = 2)), color="black", size=8) +
  scale_x_discrete(position = "top") +
  scale_y_discrete(position = "right") + 
  ggtitle("Meidan paused Pol II half-life (min)") +
  theme(panel.border = element_rect( colour = "black", fill = NA, size=1))
}

pairwise_df <- generate_pairwise_df(info_list)
pairwise_df$motif1 <- 
  factor(pairwise_df$motif1, levels = c("TATA", "Pausing", "Inr.non.G", "Inr.G"))
pairwise_df$motif2 <- 
  factor(pairwise_df$motif2, levels = rev(c("TATA", "Pausing", "Inr.non.G", "Inr.G")))

promoter_number_heatmap(pairwise_df)
half_life_heatmap(pairwise_df)
```

### Non-Mutually exclusive model

```{r inr_differences_boxplot_non_mutually_exclusive, fig.height=6, fig.width=8}

tata <- subset(new_info_df, TATA) %>%
  data.frame(type = "TATA")
tata_inr_non_g <- subset(new_info_df, TATA & Inr_non_G  )%>%
  data.frame(type = "TATA Inr non G")
tata_inr_g <- subset(new_info_df, TATA & Inr_G)%>%
  data.frame(type = "TATA Inr G")

tata_pausing <- subset(new_info_df, TATA & pausing_elements )%>%
  data.frame(type = "TATA pausing")
tata_pausing_inr_non_g <- subset(new_info_df, TATA & pausing_elements & Inr_non_G) %>%
  data.frame(type = "TATA pausing Inr non G")
tata_pausing_inr_g <- subset(new_info_df, TATA & pausing_elements & Inr_G)%>%
  data.frame(type = "TATA pausing Inr G")

pausing <- subset(new_info_df,  pausing_elements)%>%
  data.frame(type = "pausing")
pausing_inr_non_g <- subset(new_info_df,  pausing_elements & Inr_non_G )%>%
  data.frame(type = "pausing Inr non G")
pausing_inr_g <- subset(new_info_df,  pausing_elements & Inr_G)%>%
  data.frame(type = "pausing Inr G")

inr_non_g <- subset(new_info_df,  Inr_non_G) %>%
  data.frame(type = "Inr non G")
inr_g <- subset(new_info_df,  Inr_G ) %>%
  data.frame(type = "Inr G")

all_combined <- rbind(tata,pausing, inr_non_g, inr_g, tata_pausing, tata_inr_non_g, 
                      pausing_inr_non_g, tata_pausing_inr_non_g, 
                      tata_inr_g, pausing_inr_g, tata_pausing_inr_g)

half_life_boxplot(all_combined, 
        "Inr variants and paused Pol II half-life \n non-mutually exclusive")

pairwise_type <- unique(all_combined$type) %>% 
  as.character(.) %>%
  combn(., 2, simplify = F)

pval_df <- lapply(pairwise_type, function(x)calc_pval(x[1], x[2], all_combined)) %>%
  do.call(rbind, .)
pander(pval_df)

write.csv(pval_df, file = "Inr_variant_halflife_wilcox_test_non_mutually_exclusive.csv")
```

```{r inr_differences_heatmap_non_mutually_exclusive, fig.height=6, fig.width=8}
info_list <- list(TATA_TATA = tata,
                  Inr.non.G_Inr.non.G = inr_non_g,
                  Inr.G_Inr.G = inr_g,
                  Pausing_Pausing = pausing,
                  TATA_Inr.non.G = tata_inr_non_g,
                  TATA_Inr.G = tata_inr_g,
                  Pausing_Inr.non.G = pausing_inr_non_g,
                  Pausing_Inr.G = pausing_inr_g,
                  TATA_Pausing = tata_pausing)

pairwise_df <- generate_pairwise_df(info_list)
pairwise_df$motif1 <- 
  factor(pairwise_df$motif1, levels = c("TATA", "Pausing", "Inr.non.G", "Inr.G"))
pairwise_df$motif2 <- 
  factor(pairwise_df$motif2, levels = rev(c("TATA", "Pausing", "Inr.non.G", "Inr.G")))

promoter_number_heatmap(pairwise_df)
half_life_heatmap(pairwise_df)

```

# Session Info

This analysis was performed with the following R/Bioconductor session:

```{r}
sessionInfo()
```