---
title: "Figure S5 Reporter-ChIP-nexus recapitulates gene-specific Pol II pausing stability"
author: "Wanqing Shao(was@stowers.org) "
output:
  pdf_document:
    toc: yes
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancypagestyle{plain}{\pagestyle{fancy}}
---

# Description

To test if gene-specific paused Pol II stability is also recapitulated on the reporter, we treated the transfected Kc167 cell and *Drosophila pseudoobscura* ML83-63 cells with DMSO or Triptolide (TRI). TRI blocks transcription initiation, leading to the loss of Pol II signal at the pausing position. The degree of Pol II loss at the pausing position is propotional to the stability of paused Pol II.

# Enviroment setup

``` {r setup, message=FALSE, error=FALSE}
library(GenomicRanges, warn.conflicts=F)
library(magrittr)

setwd("/data/analysis_code")
options(knitr.figure_dir = 
          "FigureS5_reporter_chipnexus_recapitulates_gene_specific_polii_pausing_stability"
        )

source("shared_code/knitr_common.r")
source("shared_code/ggplot_common.r")
source("shared_code/granges_common.r")
source("shared_code/metapeak_common.r")
source("shared_code/sample_common.r")
```

# Analysis

## Reporter-ChIP-nexus recapitulates gene-specific Pol II pausing stability

```{r dmso_vs_tri, fig.width = 8, fig.height= 6}

plasmid_annotations <- import("./plasmid_annotation.bed")
genome_annotations <- import("./dps_genome_annotation.bed")

dps_dmso_path <- load_bigwig("genome_dps_dmso_1h_rpb3_chipnexus")
dps_tri_path <- load_bigwig("genome_dps_triptolide_1h_rpb3_chipnexus")

get_exo_metapeak <- function(sample, upstream=100, downstream=101, 
                             smooth=NA, endogeneous = F, dps_sample_path=NULL){
	
  gene <- as.character(subset(sample_list, sample_name == sample)$gene)
	chromosome <- as.character(subset(sample_list, sample_name == sample )$chromosome)
	
	if(endogeneous ==F){
	  
	  	sample_path <- load_bigwig(sample)
	  	region <- plasmid_annotations[seqnames(plasmid_annotations) == chromosome & 
	  	                              plasmid_annotations$name == gene] %>%
	  	          resize(., 1, "start")
	    seqlevels(region) <- chromosome
      metapeak <- exo_metapeak(region, sample_path, 
                               upstream=upstream, downstream=downstream, 
                               sample_name=gene, smooth=smooth)
      metapeak$sample <- paste(metapeak$sample_name, metapeak$strand)
      metapeak
      
	}else{
	  
  	  region <- genome_annotations[grep(gene, genome_annotations$name, ignore.case = T)]
  	  seqlevels(region) <- as.character(seqnames(region))
  	  metapeak <- exo_metapeak(region, dps_sample_path, 
  	                           upstream=upstream, downstream=downstream, 
  	                           sample_name=gene, smooth=smooth)
  	  metapeak$sample <- paste(metapeak$sample_name, metapeak$strand)
      metapeak
	 
	}
}

compare_dmso_and_tri <- function(dmso, tri){
  
  plasmid_dmso <- get_exo_metapeak(dmso, upstream=150, downstream = 151)
  plasmid_tri <- get_exo_metapeak(tri, upstream=150, downstream = 151)
  
  plasmid_tri$reads <- plasmid_tri$reads / sum(abs(plasmid_dmso$reads))
  plasmid_dmso$reads <- plasmid_dmso$reads / sum(abs(plasmid_dmso$reads))

  genome_dmso <- get_exo_metapeak(dmso, upstream=150, downstream = 151, 
                                  endogeneous = T, dps_sample_path = dps_dmso_path)
  genome_tri <- get_exo_metapeak(tri, upstream=150, downstream = 151, 
                                 endogeneous = T, dps_sample_path = dps_tri_path)  
  
  genome_tri$reads <- genome_tri$reads / sum(abs(genome_dmso$reads))
  genome_dmso$reads <- genome_dmso$reads / sum(abs(genome_dmso$reads))
  
  plasmid_dmso$sample_name <- paste(plasmid_dmso$sample_name, "plasmid control")
  plasmid_tri$sample_name <- paste(plasmid_tri$sample_name, "plasmid TRI")
  
  genome_dmso$sample_name <- paste(genome_dmso$sample_name, "genome control")
  genome_tri$sample_name <- paste(genome_tri$sample_name, "genome TRI")
  
  metapeak_c <- rbind(plasmid_dmso, plasmid_tri, genome_dmso, genome_tri)
  
  plot <- ggplot(subset(metapeak_c, strand == "+"), 
                 aes(x = tss_distance, y = reads, fill = strand))+ 
          geom_bar(fill="#B23F49", stat="identity") + 
          geom_vline(xintercept =0, linetype = "dotdash")+
          geom_bar(data=subset(metapeak_c, strand == "-"), 
                   aes(x=tss_distance, y=reads), fill="#045CA8", stat="identity")+
          xlab("Distance from TSS (bp)")+ ylab("Normalized reads")+
          facet_wrap(facets = "sample_name", ncol =2 ) +
          theme(axis.text.x = element_text(size=13), 
                axis.text.y = element_text(size=13),
                axis.title.x   = element_text(size=13),
                axis.title.y   = element_text(size=13))
  print(plot)
}

compare_dmso_and_tri("reporter_dmso_1h_dps_comm2_rpb3_chipnexus", 
                     "reporter_triptolide_1h_dps_comm2_rpb3_chipnexus")
compare_dmso_and_tri("reporter_dmso_1h_dps_pk_rpb3_chipnexus", 
                     "reporter_triptolide_1h_dps_pk_rpb3_chipnexus")
```

# Session Info

This analysis was performed with the following R/Bioconductor session:

```{r}
sessionInfo()
```
